implementations/rust/ockam/ockam_command/src/secure_channel/listener/create.rs

```diff
@@ -33,42 +34,33 @@ pub struct SecureChannelListenerNodeOpts {
 }
 
 impl CreateCommand {
-    pub fn run(self, options: CommandGlobalOpts) {
-        let cfg = options.config;
-        let node = extract_address_value(&self.node_opts.at).unwrap_or_else(|_| "".to_string());
-        let port = cfg.get_node_port(&node).unwrap();
-
-        connect_to(port, self, |ctx, cmd, rte| async {
-            create_listener(&ctx, cmd.address, cmd.authorized_identifier, rte).await?;
-            drop(ctx);
-            Ok(())
-        });
+    pub fn run(self, opts: CommandGlobalOpts) {
+        node_rpc(rpc, (opts, self));
     }
 }
 
-pub async fn create_listener(
-    ctx: &ockam::Context,
-    addr: Address,
-    authorized_identifiers: Option<Vec<IdentityIdentifier>>,
-    mut base_route: Route,
-) -> anyhow::Result<()> {
-    let resp: Vec<u8> = ctx
-        .send_and_receive(
-            base_route.modify().append(NODEMANAGER_ADDR),
-            api::create_secure_channel_listener(&addr, authorized_identifiers)?,
-        )
-        .await?;
-
-    let response = api::parse_create_secure_channel_listener_response(&resp)?;
+async fn rpc(ctx: Context, (opts, cmd): (CommandGlobalOpts, CreateCommand)) -> crate::Result<()> {
+    run_impl(&ctx, (opts, cmd)).await
+}
 
-    match response.status() {
-        Some(Status::Ok) => {
-            println!("/service/{}", addr.address());
-            Ok(())
-        }
-        _ => {
-            eprintln!("An error occurred while creating secure channel listener",);
-            std::process::exit(exitcode::CANTCREAT)
-        }
-    }
+async fn run_impl(
+    ctx: &Context,
+    (opts, cmd): (CommandGlobalOpts, CreateCommand),
+) -> crate::Result<()> {
+    let node = extract_address_value(&cmd.node_opts.at).unwrap_or_else(|_| "".to_string());
+    let mut rpc = Rpc::background(&ctx, &opts, &node)?;
+    let req = Request::post("/node/secure_channel_listener").body(
+        CreateSecureChannelListenerRequest::new(&cmd.address, cmd.authorized_identifier),
+    );
```

Comment:
or would it be better to refactor and call [this create_secure_channel_listener function](https://github.com/build-trust/ockam/blob/5b8ffc4051d6fee08354bc5a5f26abe8ab05ad3e/implementations/rust/ockam/ockam_command/src/util/api.rs#L158-L173) instead?